pilot/pkg/networking/core/loadbalancer/loadbalancer.go

```diff
@@ -37,30 +37,24 @@ const (
 func GetLocalityLbSetting(
 	mesh *v1alpha3.LocalityLoadBalancerSetting,
 	destrule *v1alpha3.LocalityLoadBalancerSetting,
-) *v1alpha3.LocalityLoadBalancerSetting {
-	var enabled bool
-	// Locality lb is enabled if its not explicitly disabled in mesh global config
-	if mesh != nil && (mesh.Enabled == nil || mesh.Enabled.Value) {
-		enabled = true
-	}
-	// Unless we explicitly override this in destination rule
+	service *model.Service,
+) (*v1alpha3.LocalityLoadBalancerSetting, bool) {
 	if destrule != nil {
 		if destrule.Enabled != nil && !destrule.Enabled.Value {
```

Comment:
If dest rule is present, but not enabled, why don't  we fallthrough to the service and mesh logic (both before and after this PR)?

Reply:
The before is more straight forward - the DR is supposed to override the mesh level. So if you disable at the DR level it should override mesh being enabled